sim/paladin/shield_of_righteousness.go

```diff
@@ -9,6 +9,8 @@ import (
 
 func (paladin *Paladin) registerShieldOfRighteousnessSpell() {
 	baseCost := paladin.BaseMana * 0.06
+	has4pT8 := paladin.HasSetBonus(ItemSetAegisPlate, 4)
+	procAura := paladin.NewTemporaryStatsAura("Aegis", core.ActionID{SpellID: 64883}, stats.Stats{stats.BlockValue: 225}, time.Second*6)
```

Comment:
Could we refactor this slightly like so?

```
var aegisPlateProcAura *core.Aura
if paladin.HasSetBonus(ItemSetAegisPlate, 4) {
  aegisPlateProcAura = paladin.NewTemporaryStatsAura("Aegis", core.ActionID{SpellID: 64883}, stats.Stats{stats.BlockValue: 225}, time.Second*6)
}

...
...

if aegisPlateProcAura != nil {
  aegisPlateProcAura.Activate(sim)
}
```

The advantage is that the aura won't be registered unless the set bonus is equipped, so in most cases we'll have 1 less aura attached to the unit.

Reply:
Yeah that makes more sense.